Data/Student.cs

```diff
@@ -0,0 +1,18 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace StudentGrades.Data;
+
+public class Student
+{
+    public int StudentId { get; set; }
+
+    [Required(ErrorMessage = "Please enter name")]
+    public string Name { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter subject")]
+    public string Subject { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter grade")]
+    [RegularExpression("^([1-9]|10)$", ErrorMessage = "Please enter a valid number from 0 to 10")]
```

Comment:
```suggestion
    [Range(0, 10, ErrorMessage = "Please enter a valid number from 0 to 10")]
```
Во-первых, куда делся 0, раз уж в сообщении говорят, что можно вводить =)
Во-вторых, как-то очень строго - даже 01 вводить нельзя
(-0.5)